src/app/page/UserProfile/LinkPhoneAccount.jsx

```diff
@@ -19,20 +19,38 @@ function LinkPhoneAccount({ firebase, auth, data, errorHandler }) {
   const classes = useStyles();
 
   async function onPhoneClick() {
+    const successCallback = (verifier) => {
+      verifier.clear();
+      setSuccessSnackbarOpen(true);
+      window.location.reload(false);
+    };
+    const verificationPrompt =
+      "Please enter the verification code that was sent to your mobile device.";
     try {
       const verifier = new firebase.auth.RecaptchaVerifier("phone-number-link", {
         size: "invisible",
       });
 
       auth.useDeviceLanguage();
-      const confirmationResult = await auth.currentUser.linkWithPhoneNumber(phoneNumber, verifier);
-      const verificationCode = window.prompt(
-        "Please enter the verification code that was sent to your mobile device."
-      );
-      await confirmationResult.confirm(verificationCode);
-      verifier.clear();
-      setSuccessSnackbarOpen(true);
-      window.location.reload(false);
+      if (!currentUserPhoneNumber) {
+        const confirmationResult = await auth.currentUser.linkWithPhoneNumber(
+          phoneNumber,
+          verifier
+        );
+        const verificationCode = window.prompt(verificationPrompt);
+        await confirmationResult.confirm(verificationCode);
+        successCallback(verifier);
+      } else {
+        const provider = new firebase.auth.PhoneAuthProvider();
+        const verificationId = await provider.verifyPhoneNumber(phoneNumber, verifier);
+        const verificationCode = window.prompt(verificationPrompt);
+        const phoneCredential = firebase.auth.PhoneAuthProvider.credential(
+          verificationId,
+          verificationCode
+        );
+        await auth.currentUser.updatePhoneNumber(phoneCredential);
+        successCallback(verifier);
```

Comment:
could move this callback out of the if/else so it only has to be called in the one place.
Think this looks good otherwise.